core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -243,6 +243,7 @@ object DruidBeamMaker
     val tsUtc = new DateTime(ts.millis, ISOChronology.getInstanceUTC)
 
     val cycleBucket = segmentGranularity match {
+      case Granularity.SECOND => tsUtc.hourOfDay().get % 3 * 60 + tsUtc.minuteOfHour().get // 10800 buckets
```

Comment:
This should involve "secondOfMinute" somewhere in it, otherwise two seconds for the same minute would get the same bucket (which we want to avoid- tasks for nearby but distinct time ranges should have different buckets)